**KELOMPOK 13**

* Rakha Farras Syaddad (11210940000077)
* Bagas Dwi Firmansyah (11210940000080)
* Muhammad Syafri Nuryanto (11210940000079)
* Mentari Syahida (11210940000069)
* Athaya Talitha Hasna (11210940000058)

Data Understanding (DU):
Keterangan Varabel:

Created_at: Tanggal dimana iklan rumah dibuat

LT : Luas Tanah properti yang ditawarkan

LB : Luas Bangunan properti yang ditawarkan

KT : Jumlah Kamar Tidur properti yang ditawarkan

KM : Jumlah Kamar Mandi properti yang ditawarkan

Garasi : Jumlah Kapasitas Garasi di properti yang ditawarkan

Carport : Kapasitas Carport di properti yang ditawarkan

Lokasi : Detail lokasi properti yang ditawarkan

Sertifikat: Jenis sertifikat properti yang ditawarkan

Listrik : Daya listrik dari properti yang ditawarkan

Hadap : Keterangan property yang ditawarkan menghadap ke arah mana

Harga : Harga property yang ditawarkan dalam Rupiah

URL : Alamat tautan (link) property yang ditawarkan.

Deskripsi : Deskripsi iklan property yang ditawarkan.

In [1]:
import warnings
warnings.filterwarnings('ignore')
import pandas as pd
!mkdir data
!wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/bekasi.csv
!wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/bogor.csv
dfB = pd.read_csv("data/bekasi.csv", low_memory = False, encoding='utf8')
dfb = pd.read_csv("data/bogor.csv", low_memory = False, encoding='utf8')

mkdir: cannot create directory ‘data’: File exists
--2022-09-28 05:18:29--  https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/bekasi.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 71038 (69K) [text/plain]
Saving to: ‘data/bekasi.csv.10’

bekasi.csv.10       100%[===================>]  69.37K  --.-KB/s    in 0.02s   

2022-09-28 05:18:29 (3.28 MB/s) - ‘data/bekasi.csv.10’ saved [71038/71038]

--2022-09-28 05:18:30--  https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/bogor.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443.


*   dfB = Variabel yang menyatakan data dari file "bekasi"
*   dfb = Variabel yang menyatakan data dari file "bogor"

In [2]:
dfB

created_at     LT     LB   KT   KM  garasi  carport            lokasi  \
0    08-12-2021   71.0   43.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
1    08-12-2021   59.0   60.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
2    08-12-2021  124.0   95.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
3    08-12-2021  144.0  100.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
4    08-12-2021  108.0  100.0  3.0  2.0     NaN      1.0  Kabupaten Bekasi   
..          ...    ...    ...  ...  ...     ...      ...               ...   
987         NaN    NaN    NaN  NaN  NaN     NaN      NaN               NaN   
988         NaN    NaN    NaN  NaN  NaN     NaN      NaN               NaN   
989         NaN    NaN    NaN  NaN  NaN     NaN      NaN               NaN   
990         NaN    NaN    NaN  NaN  NaN     NaN      NaN               NaN   
991         NaN    NaN    NaN  NaN  NaN     NaN      NaN               NaN   

    sertifikat  listrik  ... Unnamed: 16  Unnamed: 17 Unnamed: 18 Unnamed: 19  \
0          SHM   1300.0  ...         NaN          NaN         NaN         NaN   
1          SHM      NaN  ...         NaN          NaN         NaN         NaN   
2          SHM      NaN  ...         NaN          NaN         NaN         NaN   
3          SHM   2200.0  ...         NaN          NaN         NaN         NaN   
4          SHM   2200.0  ...         NaN          NaN         NaN         NaN   
..         ...      ...  ...         ...          ...         ...         ...   
987        NaN      NaN  ...         NaN          NaN         NaN         NaN   
988        NaN      NaN  ...         NaN          NaN         NaN         NaN   
989        NaN      NaN  ...         NaN          NaN         NaN         NaN   
990        NaN      NaN  ...         NaN          NaN         NaN         NaN   
991        NaN      NaN  ...         NaN          NaN         NaN         NaN   

     Unnamed: 20  Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  \
0            NaN          NaN          NaN          NaN          NaN   
1            NaN          NaN          NaN          NaN          NaN   
2            NaN          NaN          NaN          NaN          NaN   
3            NaN          NaN          NaN          NaN          NaN   
4            NaN          NaN          NaN          NaN          NaN   
..           ...          ...          ...          ...          ...   
987          NaN          NaN          NaN          NaN          NaN   
988          NaN          NaN          NaN          NaN          NaN   
989          NaN          NaN          NaN          NaN          NaN   
990          NaN          NaN          NaN          NaN          NaN   
991          NaN          NaN          NaN          NaN          NaN   

     Unnamed: 25  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  
..           ...  
987          NaN  
988          NaN  
989          NaN  
990          NaN  
991          NaN  

[992 rows x 26 columns]

#Menghilangkan *missing* *value*

In [3]:
dfB.notnull().sum()

created_at     107
LT             107
LB             107
KT             107
KM             106
garasi          33
carport         81
lokasi         107
sertifikat     106
listrik        101
hadap           48
harga          107
URL            107
deskripsi      107
Unnamed: 14      0
Unnamed: 15      0
Unnamed: 16      0
Unnamed: 17      0
Unnamed: 18      0
Unnamed: 19      0
Unnamed: 20      0
Unnamed: 21      0
Unnamed: 22      0
Unnamed: 23      0
Unnamed: 24      0
Unnamed: 25      0
dtype: int64

In [4]:
#Menghilangkan data yang tidak diperlukan
dfB = dfB.iloc[0:107]
dfB = dfB[['created_at', 'LT','LB','KT','KM','garasi','carport','lokasi','sertifikat','listrik','hadap','harga','URL','deskripsi']]
dfB

created_at     LT     LB   KT   KM  garasi  carport            lokasi  \
0    08-12-2021   71.0   43.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
1    08-12-2021   59.0   60.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
2    08-12-2021  124.0   95.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
3    08-12-2021  144.0  100.0  2.0  1.0     NaN      1.0  Kabupaten Bekasi   
4    08-12-2021  108.0  100.0  3.0  2.0     NaN      1.0  Kabupaten Bekasi   
..          ...    ...    ...  ...  ...     ...      ...               ...   
102  01-12-2021  180.0  220.0  5.0  3.0     NaN      NaN       Kota Bekasi   
103  01-12-2021  140.0  250.0  3.0  2.0     NaN      NaN  Kabupaten Bekasi   
104  01-12-2021   60.0   55.0  2.0  1.0     NaN      NaN  Kabupaten Bekasi   
105  01-12-2021   36.0   60.0  2.0  1.0     NaN      NaN  Kabupaten Bekasi   
106  01-12-2021   60.0   55.0  2.0  1.0     NaN      NaN  Kabupaten Bekasi   

    sertifikat  listrik    hadap         harga  \
0          SHM   1300.0      NaN  5.500000e+08   
1          SHM      NaN      NaN  4.400000e+08   
2          SHM      NaN      NaN  6.990000e+08   
3          SHM   2200.0  Selatan  1.350000e+09   
4          SHM   2200.0    Timur  1.400000e+09   
..         ...      ...      ...           ...   
102        SHM   2200.0      NaN  1.600000e+09   
103        SHM   2200.0  Selatan  8.500000e+08   
104        SHM   1300.0      NaN  2.500000e+08   
105        SHM   1300.0    Timur  3.000000e+08   
106        SHM   1300.0      NaN  2.500000e+08   

                                                   URL  \
0    Rumah murah di metland cibitung bekasi - Dijua...   
1    Rumah dijual di Griya gandasari cibitung - Dij...   
2    Jual cepat rumah huuk di metland cibitung beka...   
3    https://www.realoka.com/216243-rumah-rapi-siap...   
4    https://www.realoka.com/207964-jual-rumah-bagu...   
..                                                 ...   
102  https://www.rumah123.com/properti/bekasi/hos92...   
103  https://www.rumah123.com/properti/bekasi/hos92...   
104  https://www.rumah123.com/properti/bekasi/hos92...   
105  https://www.rumah123.com/properti/bekasi/hos92...   
106  https://www.rumah123.com/properti/bekasi/hos92...   

                                             deskripsi  
0    Selangkah ke stasiun krl telaga murni, Dekat d...  
1    Rumah sangat nyaman,dan terbukti bebas banjir,...  
2    Bebas banjir, Carport dan tanah luas ada taman...  
3    (Harapan Indah) Cluster Aralia Kota Harapan In...  
4    (Harapan Indah) Cluster Harmoni Perumahan Kota...  
..                                                 ...  
102  Akses pintu tol Bekasibarat; Dekat Mall Summar...  
103  DIJUAL RUMAH, Taman Kebalen bekasi cluster, Ju...  
104  Di jual Rumah Lelang Bank Pondok Ungu Permai, ...  
105  Lokasi Super Strategis, Selangkah menuju Kota ...  
106  RCG/20/V/010 MP/LB/THM/010 Di jual Rumah Lelan...  

[107 rows x 14 columns]

In [5]:
#Menampilkan data yang sudah dibersihkan
dfB = dfB.dropna()
dfB

created_at     LT     LB   KT   KM  garasi  carport            lokasi  \
17  06-12-2021  200.0  250.0  4.0  3.0     1.0      1.0       Kota Bekasi   
18  06-12-2021   60.0   55.0  2.0  1.0     1.0      1.0       Kota Bekasi   
24  06-12-2021   60.0   40.0  2.0  1.0     1.0      1.0       Kota Bekasi   
33  06-12-2021  200.0  250.0  4.0  3.0     1.0      1.0       Kota Bekasi   
38  06-12-2021   54.0   84.0  2.0  1.0     1.0      1.0  Kabupaten Bekasi   
39  05-12-2021   90.0   80.0  2.0  1.0     1.0      1.0       Kota Bekasi   
46  05-12-2021  112.0  120.0  3.0  2.0     1.0      1.0  Kabupaten Bekasi   
49  05-12-2021  100.0  112.0  2.0  2.0     1.0      1.0       Kota Bekasi   
52  05-12-2021  101.0   90.0  3.0  3.0     1.0      1.0       Kota Bekasi   
53  05-12-2021  100.0  112.0  2.0  2.0     1.0      1.0       Kota Bekasi   
61  04-12-2021   84.0  180.0  4.0  3.0     2.0      2.0       Kota Bekasi   
65  04-12-2021   60.0   32.0  2.0  1.0     1.0      1.0  Kabupaten Bekasi   
71  04-12-2021  160.0   70.0  3.0  1.0     1.0      1.0  Kabupaten Bekasi   
90  02-12-2021  135.0  110.0  3.0  3.0     1.0      1.0       Kota Bekasi   
91  02-12-2021   60.0   60.0  2.0  2.0     1.0      1.0  Kabupaten Bekasi   
94  02-12-2021  105.0  130.0  3.0  3.0     1.0      1.0       Kota Bekasi   

   sertifikat  listrik    hadap         harga  \
17        SHM   2200.0  Selatan  2.300000e+09   
18        SHM   1300.0    Barat  6.000000e+08   
24        SHM   1300.0    Timur  3.750000e+08   
33        SHM   2200.0  Selatan  2.300000e+09   
38        SHM   1300.0    Timur  6.500000e+08   
39        SHM   2200.0  Selatan  8.900000e+08   
46        SHM   2200.0  Selatan  1.100000e+09   
49        SHM   1300.0  Selatan  7.500000e+08   
52        SHM   2200.0  Selatan  1.600000e+09   
53        SHM   1300.0  Selatan  7.500000e+08   
61        SHM   2200.0    Utara  1.350000e+09   
65        SHM   1300.0    Timur  4.810000e+08   
71        SHM    900.0  Selatan  9.250000e+08   
90        SHM   2200.0  Selatan  1.200000e+09   
91        SHM   1300.0  Selatan  3.990000e+08   
94        SHM   2200.0  Selatan  1.200000e+09   

                                                  URL  \
17  https://www.rumah123.com/properti/bekasi/hos92...   
18  https://www.rumah123.com/properti/bekasi/hos92...   
24  https://www.rumah123.com/properti/bekasi/hos92...   
33  https://www.rumah123.com/properti/bekasi/hos92...   
38  https://www.rumah123.com/properti/bogor/hos929...   
39  https://www.rumah123.com/properti/bekasi/hos92...   
46  https://www.rumah123.com/properti/bekasi/hos92...   
49  https://www.rumah123.com/properti/bekasi/hos92...   
52                   https://rumahdijual.com/11635978   
53  Rumah Cantik Siap Huni Tanah Luas Akses 2 Mobi...   
61  https://www.rumah123.com/properti/bekasi/hos92...   
65  https://www.rumah123.com/properti/bekasi/hos92...   
71  https://www.rumah123.com/properti/bekasi/hos92...   
90  https://www.rumah123.com/properti/bekasi/hos92...   
91  https://www.rumah123.com/properti/bekasi/hos92...   
94  https://www.rumah123.com/properti/bekasi/hos92...   

                                            deskripsi  
17  rumah mewah dalam cluster eksklusif di harapan...  
18  Dijual Rumah Pesona Anggrek Harapan Bekasi Uta...  
24  Dekat jalan raya, Dekat fasilitas umum seperti...  
33  Dijual cepat harga BU !!, rumah mewah dalam cl...  
38  terletak dalam 2 wilayah berdekatan kabupaten ...  
39  Lokasi strategis aman asri tenang, security 24...  
46  Dijual rumah 2 lantai siap huni dalam cluster ...  
49  Lingkungan perumahan komplek, lokasi strategis...  
52  Luas Tanah 101m2 (6,5x7), luas Bangunan 90m2, ...  
53  Rumah Di Dukuh Zamrud Mustika Jaya, Bekasi kot...  
61  Perumahan TYTYAN KENCANA Lokasi dikawasan yg s...  
65  Smart Home Sistem: Smart Door Lock, CCTV, Fasi...  
71  Dalam komplek yg nyaman, jalan 2 mbl, butuh re...  
90  Di Jual Rumah Murah di Jaka sampurna Bekasi ,l...  
91  Lokasi Perumahan sangat Strategis dan hidup., 

In [6]:
dfB.isna().sum()

created_at    0
LT            0
LB            0
KT            0
KM            0
garasi        0
carport       0
lokasi        0
sertifikat    0
listrik       0
hadap         0
harga         0
URL           0
deskripsi     0
dtype: int64

In [7]:
dfB.duplicated().sum()

0

In [8]:
dfb

created_at    LT    LB   KT   KM  garasi  carport          lokasi  \
0    03-12-2021  75.0  38.0  2.0  1.0     1.0      1.0      Bojonggede   
1    06-12-2021  60.0  40.0  1.0  1.0     0.0      1.0  Babakan Madang   
2    06-12-2021  60.0  50.0  2.0  2.0     0.0      1.0  Babakan Madang   
3    06-12-2021  60.0  70.0  3.0  2.0     0.0      1.0  Babakan Madang   
4    06-12-2021  90.0  60.0  3.0  2.0     0.0      2.0  Babakan Madang   
..          ...   ...   ...  ...  ...     ...      ...             ...   
994         NaN   NaN   NaN  NaN  NaN     NaN      NaN             NaN   
995         NaN   NaN   NaN  NaN  NaN     NaN      NaN             NaN   
996         NaN   NaN   NaN  NaN  NaN     NaN      NaN             NaN   
997         NaN   NaN   NaN  NaN  NaN     NaN      NaN             NaN   
998         NaN   NaN   NaN  NaN  NaN     NaN      NaN             NaN   

    sertifikat  listrik  ... Unnamed: 16  Unnamed: 17 Unnamed: 18 Unnamed: 19  \
0          SHM   1300.0  ...         NaN          NaN         NaN         NaN   
1          SHM   2200.0  ...         NaN          NaN         NaN         NaN   
2          SHM   2200.0  ...         NaN          NaN         NaN         NaN   
3          SHM   2200.0  ...         NaN          NaN         NaN         NaN   
4          SHM   2200.0  ...         NaN          NaN         NaN         NaN   
..         ...      ...  ...         ...          ...         ...         ...   
994        NaN      NaN  ...         NaN          NaN         NaN         NaN   
995        NaN      NaN  ...         NaN          NaN         NaN         NaN   
996        NaN      NaN  ...         NaN          NaN         NaN         NaN   
997        NaN      NaN  ...         NaN          NaN         NaN         NaN   
998        NaN      NaN  ...         NaN          NaN         NaN         NaN   

     Unnamed: 20  Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  \
0            NaN          NaN          NaN          NaN          NaN   
1            NaN          NaN          NaN          NaN          NaN   
2            NaN          NaN          NaN          NaN          NaN   
3            NaN          NaN          NaN          NaN          NaN   
4            NaN          NaN          NaN          NaN          NaN   
..           ...          ...          ...          ...          ...   
994          NaN          NaN          NaN          NaN          NaN   
995          NaN          NaN          NaN          NaN          NaN   
996          NaN          NaN          NaN          NaN          NaN   
997          NaN          NaN          NaN          NaN          NaN   
998          NaN          NaN          NaN          NaN          NaN   

     Unnamed: 25  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  
..           ...  
994          NaN  
995          NaN  
996          NaN  
997          NaN  
998          NaN  

[999 rows x 26 columns]

In [9]:
dfb.notnull().sum()

created_at     193
LT             193
LB             193
KT             193
KM             193
garasi         193
carport        193
lokasi         193
sertifikat     193
listrik        193
hadap           19
harga          193
URL            193
deskripsi      193
Unnamed: 14      0
Unnamed: 15      0
Unnamed: 16      0
Unnamed: 17      0
Unnamed: 18      0
Unnamed: 19      0
Unnamed: 20      0
Unnamed: 21      0
Unnamed: 22      0
Unnamed: 23      0
Unnamed: 24      0
Unnamed: 25      0
dtype: int64

In [10]:
#Menghilangkan data yang tidak diperlukan
dfb = dfb.iloc[0:193]
dfb = dfb[['created_at', 'LT','LB','KT','KM','garasi','carport','lokasi','sertifikat','listrik','hadap','harga','URL','deskripsi']]
dfb

created_at     LT     LB   KT   KM  garasi  carport          lokasi  \
0    03-12-2021   75.0   38.0  2.0  1.0     1.0      1.0      Bojonggede   
1    06-12-2021   60.0   40.0  1.0  1.0     0.0      1.0  Babakan Madang   
2    06-12-2021   60.0   50.0  2.0  2.0     0.0      1.0  Babakan Madang   
3    06-12-2021   60.0   70.0  3.0  2.0     0.0      1.0  Babakan Madang   
4    06-12-2021   90.0   60.0  3.0  2.0     0.0      2.0  Babakan Madang   
..          ...    ...    ...  ...  ...     ...      ...             ...   
188  03-12-2021   90.0   36.0  2.0  1.0     0.0      1.0    Gunung Putri   
189  03-12-2021  107.0   31.0  2.0  1.0     0.0      1.0        Sukaraja   
190  03-12-2021  210.0  150.0  4.0  2.0     1.0      1.0   Bogor Selatan   
191  03-12-2021   90.0   36.0  2.0  1.0     0.0      1.0       Cileungsi   
192  03-12-2021   72.0   90.0  3.0  2.0     1.0      1.0        Cibinong   

    sertifikat  listrik  hadap         harga  \
0          SHM   1300.0  Barat  6.925500e+08   
1          SHM   2200.0    NaN  8.114000e+08   
2          SHM   2200.0    NaN  9.005000e+08   
3          SHM   2200.0    NaN  1.001900e+09   
4          SHM   2200.0    NaN  1.200800e+09   
..         ...      ...    ...           ...   
188        SHM   1300.0    NaN  4.405500e+08   
189        SHM   1300.0    NaN  4.086500e+08   
190        SHM   2200.0    NaN  1.200000e+09   
191        SHM   1300.0    NaN  4.405500e+08   
192        SHM   2200.0    NaN  9.000000e+08   

                                                   URL  \
0    https://rumahdijual.com/bekasi/10799791-rumah-...   
1    https://www.rumah123.com/perumahan-baru/proper...   
2    https://www.rumah123.com/perumahan-baru/proper...   
3    https://www.rumah123.com/perumahan-baru/proper...   
4    https://www.rumah123.com/perumahan-baru/proper...   
..                                                 ...   
188  https://www.rumah.com/listing-properti/dijual-...   
189  https://www.rumah.com/listing-properti/dijual-...   
190  https://www.rumah.com/listing-properti/dijual-...   
191  https://www.rumah.com/listing-properti/dijual-...   
192  https://www.rumah.com/listing-properti/dijual-...   

                                             deskripsi  
0    Dekat tol jati warna, dekat sutle busway, deka...  
1    Destinasi menarik di kawasan Babakan Madang; G...  
2    Destinasi menarik di kawasan Babakan Madang; G...  
3    Destinasi menarik di kawasan Babakan Madang; G...  
4    Destinasi menarik di kawasan Babakan Madang; G...  
..                                                 ...  
188      Dekat taman buah mekarsari, dekat rumah sakit  
189                                    Rumah siap huni  
190                              Cuma 10 menit ke kota  
191                     Dekat rumah sakit, dekat pasar  
192                        Nyaman, aman, swimming pool  

[193 rows x 14 columns]

In [11]:
#Menampilkan data yang sudah dibersihkan
dfb = dfb.dropna()
dfb

created_at     LT     LB   KT   KM  garasi  carport          lokasi  \
0    03-12-2021   75.0   38.0  2.0  1.0     1.0      1.0      Bojonggede   
7    06-12-2021   97.0   72.0  3.0  2.0     0.0      1.0        Cibinong   
10   06-12-2021  312.0   70.0  4.0  2.0     1.0      1.0    Tanah Sareal   
12   06-12-2021  112.0  129.0  3.0  3.0     0.0      0.0    Gunung Putri   
13   06-12-2021   60.0   50.0  2.0  1.0     0.0      0.0  Babakan Madang   
46   06-12-2021  167.0  300.0  4.0  3.0     2.0      0.0    Gunung Putri   
50   06-12-2021  200.0  202.0  5.0  5.0     2.0      2.0    Gunung Putri   
52   06-12-2021  162.0  164.0  5.0  5.0     0.0      2.0    Gunung Putri   
57   07-12-2021  112.0  169.0  5.0  5.0     0.0      2.0    Gunung Putri   
64   07-12-2021  120.0  160.0  5.0  3.0     0.0      1.0       Cileungsi   
95   02-12-2021   72.0   65.0  2.0  1.0     0.0      1.0         Jonggol   
96   02-12-2021   60.0   72.0  2.0  1.0     0.0      1.0         Jonggol   
97   02-12-2021   60.0   72.0  2.0  1.0     0.0      1.0       Cileungsi   
104  02-12-2021  106.0   54.0  3.0  2.0     0.0      2.0          Ciomas   
105  02-12-2021   72.0   36.0  2.0  1.0     1.0      1.0      Bojonggede   
106  02-12-2021   72.0   55.0  2.0  1.0     0.0      1.0       Cileungsi   
110  02-12-2021   62.0   45.0  2.0  1.0     1.0      1.0        Cibinong   
111  02-12-2021  698.0  350.0  4.0  4.0     2.0      2.0  Babakan Madang   
112  02-12-2021   72.0   55.0  2.0  1.0     1.0      1.0       Cileungsi   

    sertifikat  listrik           hadap         harga  \
0          SHM   1300.0           Barat  6.925500e+08   
7          SHM   2200.0         Selatan  1.230000e+09   
10         SHM   2200.0           Timur  2.500000e+08   
12        SHGB   3500.0         Selatan  1.800000e+09   
13         SHM   2200.0         Selatan  8.500000e+08   
46         SHM   5500.0    Timur, Utara  3.200000e+09   
50         SHM   4400.0  Selatan, Utara  3.128000e+09   
52         SHM   3500.0           Utara  2.536000e+09   
57         SHM   3500.0           Danau  2.128000e+09   
64         SHM   2200.0           Barat  1.250000e+09   
95        SHGB   1300.0           Utara  3.250000e+08   
96        SHGB   1300.0           Timur  2.150000e+08   
97        SHGB   1300.0           Timur  2.150000e+08   
104       SHGB   1300.0           Timur  9.090000e+08   
105       SHGB   1300.0           Timur  4.000000e+08   
106       SHGB   1300.0         Selatan  2.950000e+08   
110        SHM   1300.0           Utara  4.200000e+08   
111        SHM   5500.0         Selatan  6.850000e+09   
112       SHGB   1300.0           Barat  2.950000e+08   

                                                   URL  \
0    https://rumahdijual.com/bekasi/10799791-rumah-...   
7    https://www.rumah123.com/properti/bogor/hos920...   
10   https://www.rumah123.com/properti/bogor/hos892...   
12   https://www.rumah123.com/properti/bogor/hos820...   
13   https://www.rumah123.com/properti/bogor/hos922...   
46   https://www.rumah.com/listing-properti/dijual-...   
50   https://www.rumah.com/listing-properti/dijual-...   
52   https://www.rumah.com/listing-properti/dijual-...   
57   https://www.rumah.com/listing-properti/dijual-...   
64   https://www.rumah.com/listing-properti/dijual-...   
95   https://www.rumah123.com/properti/bogor/hos927...   
96   https://www.rumah123.com/properti/bogor/hos927...   
97   https://www.rumah123.com/properti/bogor/hos927...   
104  https://www.rumah123.com/properti/bogor/hos927...   
105  https://www.rumah123.com/properti/bogor/hos927...   
106  https://www.rumah123.com/properti/bogor/hos927...   
110  https://www.rumah123.com/properti/bogor/hos927...   
111  https://www.rumah123.com/properti/bogor/hos927...   
112  https://www.rumah123.com/properti/bogor/hos927...   

                                             deskripsi  
0    Dekat tol jati warna, dekat sutle busway, deka...  
7    Berada dalam kawasan Pemda,  Hanya 5menit ke C...  
10   Dekat 

In [12]:
dfb.isna().sum()

created_at    0
LT            0
LB            0
KT            0
KM            0
garasi        0
carport       0
lokasi        0
sertifikat    0
listrik       0
hadap         0
harga         0
URL           0
deskripsi     0
dtype: int64

In [13]:
dfb.duplicated().sum()

0

In [14]:
#Menggabungkan data 'bekasi' dan 'bogor
dfx = pd.concat([dfB, dfb])
dfx

created_at     LT     LB   KT   KM  garasi  carport            lokasi  \
17   06-12-2021  200.0  250.0  4.0  3.0     1.0      1.0       Kota Bekasi   
18   06-12-2021   60.0   55.0  2.0  1.0     1.0      1.0       Kota Bekasi   
24   06-12-2021   60.0   40.0  2.0  1.0     1.0      1.0       Kota Bekasi   
33   06-12-2021  200.0  250.0  4.0  3.0     1.0      1.0       Kota Bekasi   
38   06-12-2021   54.0   84.0  2.0  1.0     1.0      1.0  Kabupaten Bekasi   
39   05-12-2021   90.0   80.0  2.0  1.0     1.0      1.0       Kota Bekasi   
46   05-12-2021  112.0  120.0  3.0  2.0     1.0      1.0  Kabupaten Bekasi   
49   05-12-2021  100.0  112.0  2.0  2.0     1.0      1.0       Kota Bekasi   
52   05-12-2021  101.0   90.0  3.0  3.0     1.0      1.0       Kota Bekasi   
53   05-12-2021  100.0  112.0  2.0  2.0     1.0      1.0       Kota Bekasi   
61   04-12-2021   84.0  180.0  4.0  3.0     2.0      2.0       Kota Bekasi   
65   04-12-2021   60.0   32.0  2.0  1.0     1.0      1.0  Kabupaten Bekasi   
71   04-12-2021  160.0   70.0  3.0  1.0     1.0      1.0  Kabupaten Bekasi   
90   02-12-2021  135.0  110.0  3.0  3.0     1.0      1.0       Kota Bekasi   
91   02-12-2021   60.0   60.0  2.0  2.0     1.0      1.0  Kabupaten Bekasi   
94   02-12-2021  105.0  130.0  3.0  3.0     1.0      1.0       Kota Bekasi   
0    03-12-2021   75.0   38.0  2.0  1.0     1.0      1.0        Bojonggede   
7    06-12-2021   97.0   72.0  3.0  2.0     0.0      1.0          Cibinong   
10   06-12-2021  312.0   70.0  4.0  2.0     1.0      1.0      Tanah Sareal   
12   06-12-2021  112.0  129.0  3.0  3.0     0.0      0.0      Gunung Putri   
13   06-12-2021   60.0   50.0  2.0  1.0     0.0      0.0    Babakan Madang   
46   06-12-2021  167.0  300.0  4.0  3.0     2.0      0.0      Gunung Putri   
50   06-12-2021  200.0  202.0  5.0  5.0     2.0      2.0      Gunung Putri   
52   06-12-2021  162.0  164.0  5.0  5.0     0.0      2.0      Gunung Putri   
57   07-12-2021  112.0  169.0  5.0  5.0     0.0      2.0      Gunung Putri   
64   07-12-2021  120.0  160.0  5.0  3.0     0.0      1.0         Cileungsi   
95   02-12-2021   72.0   65.0  2.0  1.0     0.0      1.0           Jonggol   
96   02-12-2021   60.0   72.0  2.0  1.0     0.0      1.0           Jonggol   
97   02-12-2021   60.0   72.0  2.0  1.0     0.0      1.0         Cileungsi   
104  02-12-2021  106.0   54.0  3.0  2.0     0.0      2.0            Ciomas   
105  02-12-2021   72.0   36.0  2.0  1.0     1.0      1.0        Bojonggede   
106  02-12-2021   72.0   55.0  2.0  1.0     0.0      1.0         Cileungsi   
110  02-12-2021   62.0   45.0  2.0  1.0     1.0      1.0          Cibinong   
111  02-12-2021  698.0  350.0  4.0  4.0     2.0      2.0    Babakan Madang   
112  02-12-2021   72.0   55.0  2.0  1.0     1.0      1.0         Cileungsi   

    sertifikat  listrik           hadap         harga  \
17         SHM   2200.0         Selatan  2.300000e+09   
18         SHM   1300.0           Barat  6.000000e+08   
24         SHM   1300.0           Timur  3.750000e+08   
33         SHM   2200.0         Selatan  2.300000e+09   
38         SHM   1300.0           Timur  6.500000e+08   
39         SHM   2200.0         Selatan  8.900000e+08   
46         SHM   2200.0         Selatan  1.100000e+09   
49         SHM   1300.0         Selatan  7.500000e+08   
52         SHM   2200.0         Selatan  1.600000e+09   
53         SHM   1300.0         Selatan  7.500000e+08   
61         SHM   2200.0           Utara  1.350000e+09   
65         SHM   1300.0           Timur  4.810000e+08   
71         SHM    900.0         Selatan  9.250000e+08   
90         SHM   2200.0         Selatan  1.200000e+09   
91         SHM   1300.0         Selatan  3.990000e+08   
94         SHM   2200.0         Selatan  1.200000e+09   
0          SHM   1300.0           Barat  6.925500e+08   
7          SHM   2200.0         Selatan  1.230000e+09   
10         SHM   2200.0           Timur  2.500000e+08   
12        SHGB   3500.0         Selatan  1.800000e+09   

In [15]:
dfx.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 35 entries, 17 to 112
Data columns (total 14 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   created_at  35 non-null     object 
 1   LT          35 non-null     float64
 2   LB          35 non-null     float64
 3   KT          35 non-null     float64
 4   KM          35 non-null     float64
 5   garasi      35 non-null     float64
 6   carport     35 non-null     float64
 7   lokasi      35 non-null     object 
 8   sertifikat  35 non-null     object 
 9   listrik     35 non-null     float64
 10  hadap       35 non-null     object 
 11  harga       35 non-null     float64
 12  URL         35 non-null     object 
 13  deskripsi   35 non-null     object 
dtypes: float64(8), object(6)
memory usage: 4.1+ KB


In [16]:
#Mencari IQR untuk menghilangkan outliers
Q1 = dfx.quantile(0.25)
Q3 = dfx.quantile(0.75)
IQR = Q3 - Q1
print(IQR)

LT         6.050000e+01
LB         9.000000e+01
KT         2.000000e+00
KM         2.000000e+00
garasi     1.000000e+00
carport    0.000000e+00
listrik    9.000000e+02
harga      1.065000e+09
dtype: float64


In [17]:
dataF = dfx[~((dfx<(Q1-1.5*IQR))|(dfx>(Q3+1.5*IQR))).any(axis=1)]
dataF.shape, dfx.shape
#dataF merupakan data yang sudah dihilangkan outliernya
#dfx merupakan data sebelum dihilangkan outliernya

((25, 14), (35, 14))

* 35 - 25 = 10
* Sebanyakan 10 data yang merupakan outlier telah dibuang dari data menggunakan metode IQR

In [18]:
dataF

created_at     LT     LB   KT   KM  garasi  carport            lokasi  \
17   06-12-2021  200.0  250.0  4.0  3.0     1.0      1.0       Kota Bekasi   
18   06-12-2021   60.0   55.0  2.0  1.0     1.0      1.0       Kota Bekasi   
24   06-12-2021   60.0   40.0  2.0  1.0     1.0      1.0       Kota Bekasi   
33   06-12-2021  200.0  250.0  4.0  3.0     1.0      1.0       Kota Bekasi   
38   06-12-2021   54.0   84.0  2.0  1.0     1.0      1.0  Kabupaten Bekasi   
39   05-12-2021   90.0   80.0  2.0  1.0     1.0      1.0       Kota Bekasi   
46   05-12-2021  112.0  120.0  3.0  2.0     1.0      1.0  Kabupaten Bekasi   
49   05-12-2021  100.0  112.0  2.0  2.0     1.0      1.0       Kota Bekasi   
52   05-12-2021  101.0   90.0  3.0  3.0     1.0      1.0       Kota Bekasi   
53   05-12-2021  100.0  112.0  2.0  2.0     1.0      1.0       Kota Bekasi   
65   04-12-2021   60.0   32.0  2.0  1.0     1.0      1.0  Kabupaten Bekasi   
71   04-12-2021  160.0   70.0  3.0  1.0     1.0      1.0  Kabupaten Bekasi   
90   02-12-2021  135.0  110.0  3.0  3.0     1.0      1.0       Kota Bekasi   
91   02-12-2021   60.0   60.0  2.0  2.0     1.0      1.0  Kabupaten Bekasi   
94   02-12-2021  105.0  130.0  3.0  3.0     1.0      1.0       Kota Bekasi   
0    03-12-2021   75.0   38.0  2.0  1.0     1.0      1.0        Bojonggede   
7    06-12-2021   97.0   72.0  3.0  2.0     0.0      1.0          Cibinong   
64   07-12-2021  120.0  160.0  5.0  3.0     0.0      1.0         Cileungsi   
95   02-12-2021   72.0   65.0  2.0  1.0     0.0      1.0           Jonggol   
96   02-12-2021   60.0   72.0  2.0  1.0     0.0      1.0           Jonggol   
97   02-12-2021   60.0   72.0  2.0  1.0     0.0      1.0         Cileungsi   
105  02-12-2021   72.0   36.0  2.0  1.0     1.0      1.0        Bojonggede   
106  02-12-2021   72.0   55.0  2.0  1.0     0.0      1.0         Cileungsi   
110  02-12-2021   62.0   45.0  2.0  1.0     1.0      1.0          Cibinong   
112  02-12-2021   72.0   55.0  2.0  1.0     1.0      1.0         Cileungsi   

    sertifikat  listrik    hadap         harga  \
17         SHM   2200.0  Selatan  2.300000e+09   
18         SHM   1300.0    Barat  6.000000e+08   
24         SHM   1300.0    Timur  3.750000e+08   
33         SHM   2200.0  Selatan  2.300000e+09   
38         SHM   1300.0    Timur  6.500000e+08   
39         SHM   2200.0  Selatan  8.900000e+08   
46         SHM   2200.0  Selatan  1.100000e+09   
49         SHM   1300.0  Selatan  7.500000e+08   
52         SHM   2200.0  Selatan  1.600000e+09   
53         SHM   1300.0  Selatan  7.500000e+08   
65         SHM   1300.0    Timur  4.810000e+08   
71         SHM    900.0  Selatan  9.250000e+08   
90         SHM   2200.0  Selatan  1.200000e+09   
91         SHM   1300.0  Selatan  3.990000e+08   
94         SHM   2200.0  Selatan  1.200000e+09   
0          SHM   1300.0    Barat  6.925500e+08   
7          SHM   2200.0  Selatan  1.230000e+09   
64         SHM   2200.0    Barat  1.250000e+09   
95        SHGB   1300.0    Utara  3.250000e+08   
96        SHGB   1300.0    Timur  2.150000e+08   
97        SHGB   1300.0    Timur  2.150000e+08   
105       SHGB   1300.0    Timur  4.000000e+08   
106       SHGB   1300.0  Selatan  2.950000e+08   
110        SHM   1300.0    Utara  4.200000e+08   
112       SHGB   1300.0    Barat  2.950000e+08   

                                                   URL  \
17   https://www.rumah123.com/properti/bekasi/hos92...   
18   https://www.rumah123.com/properti/bekasi/hos92...   
24   https://www.rumah123.com/properti/bekasi/hos92...   
33   https://www.rumah123.com/properti/bekasi/hos92...   
38   https://www.rumah123.com/properti/bogor/hos929...   
39   https://www.rumah123.com/properti/bekasi/hos92...   
46   https://www.rumah123.com/properti/bekasi/hos92...   
49   https://www.rumah123.com/properti/bekasi/hos92...   
52                    https://rumahdijual.com/11635978   
53   Rumah Cantik Siap Huni Tanah Luas Akses 2 Mobi...   
65   https://www.rumah123.com/propert

In [19]:
dataF.describe(include='all')

created_at          LT          LB         KT         KM    garasi  \
count           25   25.000000   25.000000  25.000000  25.000000  25.00000   
unique           6         NaN         NaN        NaN        NaN       NaN   
top     02-12-2021         NaN         NaN        NaN        NaN       NaN   
freq            10         NaN         NaN        NaN        NaN       NaN   
mean           NaN   94.360000   90.600000   2.520000   1.680000   0.76000   
std            NaN   41.648209   57.798501   0.822598   0.852447   0.43589   
min            NaN   54.000000   32.000000   2.000000   1.000000   0.00000   
25%            NaN   60.000000   55.000000   2.000000   1.000000   1.00000   
50%            NaN   75.000000   72.000000   2.000000   1.000000   1.00000   
75%            NaN  105.000000  112.000000   3.000000   2.000000   1.00000   
max            NaN  200.000000  250.000000   5.000000   3.000000   1.00000   

        carport       lokasi sertifikat  listrik    hadap         harga  \
count      25.0           25         25     25.0       25  2.500000e+01   
unique      NaN            6          2      NaN        4           NaN   
top         NaN  Kota Bekasi        SHM      NaN  Selatan           NaN   
freq        NaN           10         19      NaN       13           NaN   
mean        1.0          NaN        NaN   1608.0      NaN  8.343020e+08   
std         0.0          NaN        NaN    460.0      NaN  5.837546e+08   
min         1.0          NaN        NaN    900.0      NaN  2.150000e+08   
25%         1.0          NaN        NaN   1300.0      NaN  3.990000e+08   
50%         1.0          NaN        NaN   1300.0      NaN  6.925500e+08   
75%         1.0          NaN        NaN   2200.0      NaN  1.200000e+09   
max         1.0          NaN        NaN   2200.0      NaN  2.300000e+09   

                                                      URL  \
count                                                  25   
unique                                                 24   
top     https://www.rumah123.com/properti/bekasi/hos92...   
freq                                                    2   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

                                               deskripsi  
count                                                 25  
unique                                                22  
top     Fasilitas masjid, taman, jogging track, keamanan  
freq                                                   4  
mean                                                 NaN  
std                                                  NaN  
min                                                  NaN  
25%                                                  NaN  
50%                                                  NaN  
75%                                                  NaN  
max                                                  NaN

4. Rumah yang paling banyak terjual dengan karakteristik sebagai berikut,
* Dengan rata-rata luas tanah 77.0 yang terjual di tanggal 02-12-2021
* Luas tanah paling banyak terjual yaitu 60.0 di tanggal 02-12-2021
* Jenis sertifikat yang paling banyak terjual adalah SHM
* Dengan rumah paling banyak terjual menghadap ke Selatan



5. Ya, dari data statistika deskriptif sederhana, kami dapat memberikan rekomendasi rumah yang berpotensi menghasilkan keuntungan bagi perusahaan. Rumah dengan harga tertinggi terdapat di Kota Bekasi dengan harga Rp2.300000+09 dengan list no.17.

data statistika deskriptif sederhana, kami dapat memberikan rekomendasi rumah yang berpotensi menghasilkan keuntungan bagi perusahaan. Rumah dengan harga terendah terdapat di daerah Jonggol dengan harga Rp2150000e+08.

6. Informasi berharga yang kami dapatkan dari hasil prepocessing dan statistika deskriptif adalah perusahaan dapat mengatahui:

* paling banyak dijual pada tanggal 02-12-2021
* Rata-rata tanah yang dijual 94.360000
* Rata-rata harga rumah yang banyak terjual RP8.343020e+08
* Harga terendah rumah yang terjual adalah Rp2.150000e+08
* Harga tertinggi rumah yang terjual Rp2.300000e+09


